src/pages/api/auth/[...nextauth].js

```diff
@@ -90,37 +102,35 @@ export const authOptions = {
   pages: {
     signIn: "/login",
     // newUser: "/signup",
-    // error: "/login",
+    error: "/login",
   },
   events: {
     createUser: async (message) => {
-      // TODO: Problem: Google won't link a user to an account if the user is already in the database
-      // await dbConnect();
-      // // await User.deleteOne({ _id }); // delete the user automatically generated by next-auth
-      // const invitedUser = await User.findOne({ email: message.user.email, _id: { $not: { $eq: message.user.id } } });
-      // await User.updateOne({ _id: message.user.id }, { acceptedInvite: true, isActive: true, role: invitedUser.role });
-      // await User.deleteOne({ _id: invitedUser._id }); // delete the previously invited user
       await dbConnect();
-      await User.updateOne(
-        { _id: message.user.id },
-        { acceptedInvite: true, isActive: true, role: "User" },
-      );
+      const invitedUser = await getInvitedUserByEmail(message.user.email);
+
+      if (!invitedUser || !invitedUser.isActive) {
+        await User.deleteOne({ email: message.user.email });
+      }
+
+      const updateObject = {
+        role: invitedUser.role,
+        acceptedInvite: true,
+        isActive: true,
+      };
+      await updateUser(message.user.id, updateObject);
+      await updateInvitedUser(invitedUser._id, updateObject);
```

Comment:
We can call `updateInvitedUser` inside `updateUser`. This is because we always want these documents to have the same values. Also it makes updating the invited user an implicit action that we don't have to worry about.